core/state/statedb.go

```diff
@@ -603,6 +621,28 @@ func (s *StateDB) getStateObject(addr common.Address) *stateObject {
 	return obj
 }
 
+// resolveStateObject follows delegation designations to resolve a state object
+// given by the address, returning nil if the object is not found or was deleted
+// in this execution context.
+func (s *StateDB) resolveCode(addr common.Address) *stateObject {
+	obj := s.getStateObject(addr)
+	if obj == nil {
+		return nil
+	}
+	if s.witness != nil {
+		s.witness.AddCode(obj.code)
+	}
+	addr, ok := types.ParseDelegation(obj.Code())
+	if !ok {
+		return obj
+	}
+	obj = s.getStateObject(addr)
```

Comment:
Is it possible that the address pointed also contains a delegation?

for example, 

contract A has DelegateToB code;
contract B has DelegateToC code;

Resolve(A) should return code C?

Reply:
I think we mentioned on call earlier, but yes it's possible however we only resolve 1 level deep in a chain of delegations.